packages/kbn-guided-onboarding/src/components/landing_page/guide_card_footer.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiButton, EuiProgress, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { GuideId, GuideState } from '../../types';
+import { UseCase } from './use_case_card';
+
+const viewGuideLabel = i18n.translate(
+  'guidedOnboardingPackage.gettingStarted.guideCard.startGuide.buttonLabel',
+  {
+    defaultMessage: 'View guide',
+  }
+);
+
+const continueGuideLabel = i18n.translate(
+  'guidedOnboardingPackage.gettingStarted.guideCard.continueGuide.buttonLabel',
+  {
+    defaultMessage: 'Continue',
+  }
+);
+
+const completedLabel = i18n.translate(
+  'guidedOnboardingPackage.gettingStarted.guideCard.progress.completedLabel',
+  {
+    defaultMessage: 'Completed',
+  }
+);
+
+const inProgressLabel = i18n.translate(
+  'guidedOnboardingPackage.gettingStarted.guideCard.progress.inProgressLabel',
+  {
+    defaultMessage: 'In progress',
+  }
+);
+
+export interface GuideCardFooterProps {
+  guides: GuideState[];
+  useCase: UseCase;
+  activateGuide: (useCase: UseCase, guideState?: GuideState) => void;
+}
+export const GuideCardFooter = ({ guides, useCase, activateGuide }: GuideCardFooterProps) => {
+  const guideState = guides.find((guide) => guide.guideId === (useCase as GuideId));
+  const viewGuideButton = (
+    <div className="eui-textCenter">
+      <EuiButton
+        // Used for FS tracking
+        data-test-subj={`onboarding--guideCard--view--${useCase}`}
+        fill
+        onClick={() => activateGuide(useCase, guideState)}
```

Comment:
Incidentally, this is also a problem: you're creating a new function on every render cycle.